eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/EventCriteriaBuilder.java

```diff
@@ -17,50 +17,93 @@
 package org.axonframework.eventsourcing.eventstore;
 
 import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
 
+import java.util.Arrays;
 import java.util.HashSet;
 import java.util.Set;
 
-final class EventCriteriaBuilder implements EventCriteria.Builder {
+/**
+ * Builder class to construct a {@link FilteredEventCriteria} instance. The builder executes in two steps:
+ *
+ * <ol>
+ *     <li>
+ *         {@link EventCriteriaBuilderEventTypeStage The event type stage}, defining the types that this criteria
+ *          instance matches with using {@link EventCriteriaBuilderEventTypeStage#eventTypes(String...)} or {@link EventCriteriaBuilderEventTypeStage#anyEventType()}.
+ *     </li>
+ *     <li>
+ *         {@link EventCriteriaBuilderEventTypeStage The tag stage}, defining the tags that events are expected to have using
+ *          {@link EventCriteriaBuilderTagStage#withTags(Tag...)}, {@link EventCriteriaBuilderTagStage#withTags(Set)}, or
+ *          {@link EventCriteriaBuilderTagStage#withTags(String...)}.
+ *     </li>
+ * </ol>
+ * <p>
+ * The builder will return an {@link EventCriteria} instance that matches on the defined types and tags.
+ *
+ * @author Mitchell Herrijgers
+ * @see EventCriteria
+ * @since 5.0.0
+ */
+public final class EventCriteriaBuilder implements EventCriteriaBuilderEventTypeStage, EventCriteriaBuilderTagStage {
```

Comment:
I've been looking at this a bit, but I am wondering if we can make it package-private. I mean, I think it's nice if we force people through the interface and its static factory methods. Differently put, I think it's good to have a single option for building (the interface) instead of two (the interface and this implementation). 

Just not sure whether we can make it package-private.

Reply:
As we only expose the interfaces, and not this class, apparently we can! Doing it.